pulsar-common/src/main/java/com/yahoo/pulsar/common/naming/DestinationDomain.java

```diff
@@ -16,5 +16,29 @@
 package com.yahoo.pulsar.common.naming;
 
 public enum DestinationDomain {
-    persistent
+    persistent("persistent"), non_persistent("non-persistent");
```

Comment:
yes, `non-persistent` is a valid token for URL schema. We already have it at [NonPersistentTopics](https://github.com/rdhabalia/pulsar/blob/34534370d7bb71fe83175a818090f951824696f2/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/admin/NonPersistentTopics.java#L51) and tested with testcase. However, I think we can change it to `nonpersistent` for consistency as `non-persistent` is not a valid variable/enum name. So, if we don't have any concern then I will change it to `nonpersistent`